dendronhq/dendron
# Pre commit test

# internal: testing pre-commit addition to project

This is a PR to test what changes pre-commit may force (and in the future, prevent) when it comes to formatting files across the codebase. I wanted to run this to see if it would impact, negatively, the automated tests that run in the CI, too.

Changes applied:
- `prettier` formatting is forced
- Remove trailing white space from lines
- Make sure files end in a newline and only a new line (helps reduce multiple empty new lines at ends of files)
- Fix mixed line endings (this is usually only a problem for Windows-based contributors)
- Fix byte order marker (this is usually only a problem for Windows-based contributors)

There are a handful of other tests ran, which can all be seen in the `.pre-commit-config.yaml` of this PR. This will also allow for:
- If wanting to configure something like `eslint` to run checks before a commit is made, can enable
  - My experience is too limited here, so I have it commented out in the config
- Can add an additional GitHub Action that runs the pre-commit step, ensuring that a test failure will appear if pre-commit is not configured locally for a contributor

# Pull Request Checklist

You can go to [dendron pull requests](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html) to see full details for items in this checklist.

## General

### Quality Assurance
- [ ] after you submit your pull request, check the output of our [integration test](https://github.com/dendronhq/dendron/actions) and make sure all tests pass
  - NOTE: if you running MacOS or Linux, pay special attention to the Windows output and vice versa if you are developing on Windows

### Docs
- [ ] Make sure that the PR title follows our [commit style](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#commit-style)
- [ ] Please summarize the feature or impact in 1-2 lines in the PR description
- [ ] If your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

Example PR Description
```markdown
# feat: capitalize all foos

This changes capitalizes all occurrences of `foo` to `Foo`

Docs PR: <URL_TO_DOCS_PR>
```

## Special Cases

### First Time PR
- [ ] sign the [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement) which will be prompted by our github bot after you submit the PR
- [ ] add your [discord](https://discord.gg/AE3NRw9) alias in the review so that we can give you the [horticulturalist](https://wiki.dendron.so/notes/7c00d606-7b75-4d28-b563-d75f33f8e0d7.html#horticulturalist) badge in our community



### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated
